cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -28,6 +28,8 @@
 
 #include "arrow/dataset/discovery.h"
 #include "arrow/dataset/file_base.h"
+#include "arrow/dataset/parquet_encryption_config.h"
+
 #include "arrow/dataset/type_fwd.h"
 #include "arrow/dataset/visibility.h"
 #include "arrow/io/caching.h"
```

Comment:
We may need to include `arrow/util/config.h` explicitly?
Could you check whether `PARQUET_REQUIRE_ENCRYPTION` is defined or not in this file?

Reply:
I also checked this.
We don't need to include `arrow/util/config.h` explicitly because `arrow/util/config.h` is included in implicitly. (`arrow/dataset/file_base.h` -> `arrow/dataset/dataset.h` -> `arrow/util/future.h` -> `arrow/util/config.h`)